x509-ocsp/src/basic.rs

```diff
@@ -0,0 +1,268 @@
+//! Basic OCSP Response
+
+use alloc::vec::Vec;
+use const_oid::AssociatedOid;
+use core::{default::Default, option::Option};
+use der::{
+    asn1::{BitString, GeneralizedTime, Null, OctetString},
+    Choice, Decode, Enumerated, Sequence,
+};
+use spki::AlgorithmIdentifierOwned;
+use x509_cert::{
+    certificate::Certificate,
+    crl::RevokedCert,
+    ext::{pkix::CrlReason, Extensions},
+    name::Name,
+    serial_number::SerialNumber,
+    time::Time,
+};
+
+/// OCSP `Version` as defined in [RFC 6960 Section 4.1.1].
+///
+/// ```text
+/// Version ::= INTEGER { v1(0) }
+/// ```
+///
+/// [RFC 6960 Section 4.1.1]: https://datatracker.ietf.org/doc/html/rfc6960#section-4.1.1
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Enumerated)]
+#[asn1(type = "INTEGER")]
+#[repr(u8)]
+pub enum Version {
+    /// Version 1 (default)
+    V1 = 0,
+}
+
+impl Default for Version {
+    fn default() -> Self {
```

Comment:
I think this can be replaced like:
```
#[derive(Default)]
pub enum Version {
    #[default]
     V1 = 0
}
```

https://doc.rust-lang.org/std/default/trait.Default.html#enums

This came around in https://blog.rust-lang.org/2022/06/30/Rust-1.62.0.html but there should have an MSRV of 1.65 because of `x509-cert`.